fix(security): enable AUTH_RATE_LIMITED to work correctly

aminghadersohi opened the pull request:
### SUMMARY

Fixes the `AUTH_RATE_LIMITED` configuration to work correctly when set to `True`. Previously, enabling this setting would cause the application to fail during startup.

**Before this fix:**
When `AUTH_RATE_LIMITED = True` is set in `superset_config.py`, the application fails to start with:
```
AttributeError: 'NoneType' object has no attribute '__module__'
```

**After this fix:**
The application starts successfully with rate limiting properly applied to the auth view.

**Root cause:**
Flask-AppBuilder's `register_views()` method tries to apply rate limiting to `self.auth_view.blueprint`, but in certain initialization contexts (e.g., MCP service's flask_singleton), the blueprint is `None`, causing the error.

**Solution:**
1. Apply rate limiting directly in Superset's `register_views()` method after the auth view is created
2. Prevent the parent's `register_views()` from attempting to re-apply rate limiting by temporarily disabling `AUTH_RATE_LIMITED` during the `super()` call

This ensures rate limiting works correctly regardless of how many Flask app instances are created.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

N/A - Backend fix

### TESTING INSTRUCTIONS

**Before fix (reproduces error):**
1. Set `AUTH_RATE_LIMITED = True` in `superset_config.py`
2. Run: `PYTHONPATH=. superset run -p 8088`
3. Observe: Application fails with AttributeError

**After fix (works correctly):**
1. Set `AUTH_RATE_LIMITED = True` in `superset_config.py`
2. Run: `PYTHONPATH=. superset run -p 8088`
3. Observe: Application starts successfully
4. Verify rate limiting is working by making multiple POST requests to `/login/` rapidly

**MCP Service test:**
1. Set `AUTH_RATE_LIMITED = True` and `MCP_AUTH_ENABLED = True` in `superset_config.py`
2. Run: `PYTHONPATH=. superset mcp run --port 5008`
3. Observe: MCP service starts successfully

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags: None
- [ ] Changes UI: No
- [ ] Includes DB Migration: No
- [ ] Introduces new feature or API: No
- [ ] Removes existing feature or API: No

aminghadersohi committed:
fix(security): enable AUTH_RATE_LIMITED to work correctly

When AUTH_RATE_LIMITED = True is set in superset_config.py, the application
would fail to start because Flask-AppBuilder's rate limiting logic tried to
apply rate limiting to the auth view blueprint, but the blueprint was None
in certain initialization contexts (e.g., MCP service's flask_singleton).

This fix applies rate limiting directly in Superset's register_views() method
after the auth view is created, and prevents the parent's register_views()
from attempting to re-apply rate limiting by temporarily disabling the
AUTH_RATE_LIMITED config during the super() call.

The fix ensures rate limiting works correctly regardless of how many Flask
app instances are created (main app, MCP service, etc.).

sadpandajoe committed:
chore: bump fastmcp

sadpandajoe committed:
fix: recompile requirements in Linux environment

aminghadersohi committed:
Merge branch 'temp-deps-fix' into fix/auth-rate-limiting

aminghadersohi committed:
fix: add exception safety to config restoration

villebro reviewed (approved):
I can confirm this works. The temporary disabling of the rate limiting config feels slightly janky, but hey, if it works it works 🙂

aminghadersohi committed:
refactor(security): address PR review comments

- Use getattr pattern instead of hasattr + separate None check for DRYness
- Remove .get() on config since AUTH_RATE_LIMITED has default value

aminghadersohi commented on superset/security/manager.py:
```diff
@@ -2875,7 +2875,31 @@ def register_views(self) -> None:
+
+        # Temporarily disable AUTH_RATE_LIMITED to prevent parent from trying to
+        # apply rate limiting to a potentially None blueprint
+        original_auth_rate_limited = current_app.config.get("AUTH_RATE_LIMITED", False)
```
done

aminghadersohi committed:
chore: trigger CI rerun
